include/nighthawk/common/request_source_plugin.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "envoy/api/api.h"
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+
+#include "nighthawk/common/request_source.h"
+
+namespace Nighthawk {
+
+/**
+ * An interface for different RequestSource plugins.
+ * Uses a plugin-specific config proto.
+ */
+class RequestSourcePlugin : RequestSource {
```

Comment:
If RequestSourcePlugin inherit from RequestSource just to add the constructor, then we don't need to create  RequestSourcePlugin. We can directly create an implementation of RequestSource instead.

Reply:
I need this interface for the register factory mechanism. And the plugins I'm making need to be request sources to be used with the RequestSourceFactory. I might be misunderstanding your comment.